system-contracts/contracts/L1Messenger.sol

```diff
@@ -194,6 +194,13 @@ contract L1Messenger is IL1Messenger, ISystemContract {
     function publishPubdataAndClearState(
         bytes calldata _totalL2ToL1PubdataAndStateDiffs
     ) external onlyCallFromBootloader {
+        // Send hyperchain batches
+        SystemContractHelper.toL1(
```

Comment:
the pubdata that comes from the batch aggregator should become part of the current batches pubdata instead of being published via a separate log. this way we wont have to verify it separate from the main portion of pubdata